packages/@aws-cdk/cloudformation-include/test/yaml-templates.test.ts

```diff
@@ -0,0 +1,382 @@
+          "Fn::And": [
+            {
+              "Condition": "AlwaysTrueCond",
+            },
+            {
+              "Fn::Or": [
+                {
+                  "Condition": "AlwaysTrueCond",
+                },
+                {
+                  "Condition": "AlwaysTrueCond",
+                },
+              ],
+            },
+          ],
+        },
+      },
+      "Resources": {
+        "Vpc": {
+          "Type": "AWS::EC2::VPC",
+          "Properties": {
+            "CidrBlock": {
+              "Fn::If": [
+                "AlwaysTrueCond",
+                {
+                  "Fn::Cidr": [
+                    "192.168.1.1/24",
+                    2,
+                    5,
+                  ],
+                },
+                {
+                  "Fn::Cidr": [
+                    "10.0.0.0/24",
+                    "6",
+                    "5",
+                  ],
+                },
+              ],
+            },
+          },
+        },
+        "Bucket": {
+          "Type": "AWS::S3::Bucket",
+          "Properties": {
+            "BucketName": {
+              "Fn::If": [
+                "AndCond",
+                {
+                  "Fn::FindInMap": [
+                    "RegionMap",
+                    "region-1",
+                    "HVM64",
+                  ],
+                },
+                "Unreachable",
+              ],
+            },
+          },
+        },
+        "Subnet1": {
+          "Type": "AWS::EC2::Subnet",
+          "Properties": {
+            "VpcId": {
+              "Fn::If": [
+                "AlwaysTrueCond",
+                {
+                  "Fn::Split": [
+                    ",",
+                    {
+                      "Fn::ImportValue": "ImportedVpcId",
+                    },
+                  ],
+                },
+                "Unreachable",
+              ],
+            },
+            "CidrBlock": "10.0.0.0/24",
+            "AvailabilityZone": {
+              "Fn::Select": [
+                "0",
+                {
+                  "Fn::GetAZs": "",
+                },
+              ],
+            },
+          },
+        },
+        "Subnet2": {
+          "Type": "AWS::EC2::Subnet",
+          "Properties": {
+            "VpcId": {
+              "Fn::Select": [
+                0,
+                {
+                  "Fn::Cidr": [
+                    "10.0.0.0/24",
+                    5,
+                    2,
+                  ],
+                },
+              ],
+            },
+            "CidrBlock": "10.0.0.0/24",
+            "AvailabilityZone": {
+              "Fn::Select": [
+                "0",
+                {
+                  "Fn::GetAZs": "eu-west-2",
+                },
+              ],
+            },
+          },
+        },
+        "TransformBucket": {
+          "Type": "AWS::S3::Bucket",
+          "Properties": {
+            "BucketName": {
+              "Fn::If": [
```

Comment:
`Fn::if` takes 3 arguments.... how does this pass tests?

Reply:
Good catch. I made the same error twice, by omitting the 3rd argument in both the yaml template and the test file

Reply:
I'm not sure how the two argument if statement is accepted though

Reply:
It's because `value[2]` returns undefined if `value` is a 2-element array... oh JavaScript 🙂. Not the biggest deal, but we should add the third argument here.